untrack: remove '--all' option

slack-coder opened the pull request:
Reduce redundancy as 'rad untrack --all' is synonymous with 'rad rm'.

Related to issue #189

erak commented on rm/src/lib.rs:
```diff
@@ -84,16 +86,31 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
         .join(options.urn.encode_id());
 
-    if !options.prompt
+    if options.prompt
```
Wouldn't this lead to the project being removed when the user does not confirm (presses `n`)?

erak replied:
Sorry, I fear the `fixup!` commit is also not doing the right thing. So right now, if the user confirms the removal, nothing happens. Is there a specific reason you changed the logic here?

slack-coder replied:
First of all, thanks for spotting this!

Yes the fixup was wrong as well.  The logic should remain the same, trying to improve the code by isolating the user confirmation to a few lines.

erak commented on rm/src/lib.rs:
```diff
@@ -84,16 +86,31 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
     }
 
+    // A project is only considered to exist if it is tracked one or more times.
+    tracking::untrack_all(
```
Mmh, is this equivalent to calling `rad_untrack::execute` with `peer == None`?

slack-coder replied:
Is there something wrong with the implementation?  This is part of the intent of the PR.

erak replied:
As far as I understand the comments in https://github.com/radicle-dev/radicle-cli/issues/189, this PR should only mostly remove the `--all` option from `rad-untrack`. And thus wondering, what the reason to replace `rad_untrack::execute()` with `tracking::untrack_all()` was. Just don't know if is supposed to behave the same or if's different for the very same reason that I don't fully get yet :wink:

slack-coder replied:
It should be ;)  calling rad_untrack::execute with peer = None is the internal equivalent of 'rad untrack --all'.  I was confused whether a problem was spotted or if it was curiosity.

Check the changes at https://github.com/radicle-dev/radicle-cli/pull/222/files#diff-48f638317ddb6a247ea94a4bca3e12f78ade01dc2233a6e818e2fef066abd524L109-L116

erak replied:
Thanks for the explanation!

erak reviewed (dismissed):
LGTM

cloudhead commented:
Needs a rebase since the `rm` changes have been merged.

slack-coder committed:
untrack: remove '--all' option

Reduce redundancy as 'rad untrack --all' is synonymous with 'rad rm'.

slack-coder committed:
rad-rm: warn on failure to remove remote

Consider failing to remove a remote a non-fatal error.  The logic is there
to assist users, not cause problems by halting further execution.

Signed-off-by: Slack Coder <slackcoder@server.ky>

slack-coder commented:
@erak @cloudhead the code was re-implemented due to the amount of recent changes.

There is an additional change to [report an error instead of failing](https://github.com/radicle-dev/radicle-cli/pull/222/commits/ca8838389c24dd75d5f9c7cfe7ffe4719cb6f90e).  There is a bug in the current code base which triggers this and is not trivial to remove.

erak commented on rm/src/lib.rs:
```diff
@@ -122,16 +124,38 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+                        let err =
+                            term::remote::remove(&p.remote.to_string(), &storage, &cwd_repo, urn);
+                        if let Err(err) = err {
+                            eprintln!("unable to remove git remote '{}': {}", p.remote, err);
```
Is this expected to fail often? And is the whole removal operation considered errornous if only one of the remotes can't be removed? If so, we could return the error here already such that it bubbles up correctly. Otherwise I'd suggest to collect the remotes that could not be removed and show an error / warning listing these.

slack-coder committed:
use term::io::error to report errors

slack-coder commented:
@erak rebased and addressed the last comment in https://github.com/radicle-dev/radicle-cli/pull/222/commits/f0bc255d342109612ab497c795dc43237af61797.

The failure to remove a subset of remotes may be different for each one, and it should not happen often.  Reporting them per line should be ok.
